### ISSUE TITLE
Why smaller batch size produces higher accuracy? 

### ISSUE THREAD

**Issue by twmht** [jwyang/faster-rcnn.pytorch#305]
from the pascal voc result, you showed that the smaller batch size produces much higher accuracy.

but this indeed not reasonable.

in the ohem paper (https://arxiv.org/abs/1604.03540), they showed that the accuracy is enhanced when batch size=2 as compared to batch size=1.

Do you have any idea?

**Comment by jwyang** [comment]
@twmht I think it is more on the training schedule tuning. As you can see, when we increase the batch size, we need also increase the number of training epochs. This is because the gradients are mixed up and the updating steps become much less compared with when batch size = 1. A bunch of papers also showed that using a large batch size will slow the convergence speed and the performance generally degrade a bit within limited training time, but finally still reach the optimal: https://stats.stackexchange.com/questions/164876/tradeoff-batch-size-vs-number-of-iterations-to-train-a-neural-network. Also, in your mentioned test case, the batch size is merely 2, which is not comparable to our training settings.

**Comment by twmht** [comment]
ok.
